### ISSUE TITLE
Use lru_cache from Python 3.2?

### ISSUE THREAD

**Issue by jsivak** [gsakkis/memoized#2]
Have you considered adding support for using functools.lru_cache() (introduced in Python 3.2)?
It requires hashable function args, so it may be somewhat limited.

**Comment by gsakkis** [comment]
Yeah `lru_cache` would be a welcome addition; Python 3 was much less popular back in 2013. Hashable args are fine, there is an [optional parameter](https://github.com/gsakkis/memoized/blob/master/src/memoized.py#L32) for this specifically.
